### PR TITLE
Add support for GraalVM. Fixes #34

### LINKED ISSUE #34: Could not initialize class org.elasticsearch.client.RestClient
Hi,

I've implemented a **Micronaut** (`2.0.0.M3`) with **Elasticsearch** (`7.6.2`). Here is my project [`graalvm-quarkus-micronaut-springboot/elasticsearch/micronaut-elasticsearch`](https://github.com/ivangfr/graalvm-quarkus-micronaut-springboot/tree/master/elasticsearch/micronaut-elasticsearch)

I didn't have any problem running it in development mode (`./gradlew clean run`) or as a Docker container (non-native); except, I had problem running it in native mode.

Basically, I am able to build the Native Docker Image and spin a Docker container of it. However, when I make a request to the Rest API, I get an exception

```
Message: Could not initialize class org.elasticsearch.client.RestClient
Path Taken: new $MovieControllerDefinition$Intercepted([MovieService movieService],MovieMapper   movieMapper,BeanContext beanContext,Qualifier qualifier,Interceptor[] interceptors) --> new   MovieServiceImpl([RestHighLevelClient client],MovieMapper movieMapper,ObjectMapper objectMapper)
io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [com.  mycompany.service.MovieServiceImpl]

Message: Could not initialize class org.elasticsearch.client.RestClient
Path Taken: new $MovieControllerDefinition$Intercepted([MovieService movieService],MovieMapper   movieMapper,BeanContext beanContext,Qualifier qualifier,Interceptor[] interceptors) --> new   MovieServiceImpl([RestHighLevelClient client],MovieMapper movieMapper,ObjectMapper objectMapper)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1840)
	at io.micronaut.context.DefaultBeanContext.getScopedBeanForDefinition(DefaultBeanContext.java:2309)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2224)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2196)
...
```
> For complete stack trace see [here](https://github.com/ivangfr/graalvm-quarkus-micronaut-springboot/tree/master/elasticsearch/micronaut-elasticsearch#issues)

It seems that the `class org.elasticsearch.client.RestClient` bean is not been initialized. I am using `micronaut-elasticsearch` dependency as you can see in the [`build.gradle`](https://github.com/ivangfr/graalvm-quarkus-micronaut-springboot/blob/master/elasticsearch/micronaut-elasticsearch/build.gradle).

Thanks

### PR DESCRIPTION
**Pull request by ilopmar**

### CODE REVIEW

File: src/main/docs/guide/graalvm.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+[source,groovy]
+----
+// Enable logging on JVM
+runtimeOnly 'org.slf4j:log4j-over-slf4j:1.7.30'
+
+// Additional dependencies for logging on GraalVM
+implementation 'org.apache.logging.log4j:log4j-api:2.13.3'
+implementation 'org.apache.logging.log4j:log4j-core:2.13.3'
+----
```
Comment (graemerocher) [inline src/main/docs/guide/graalvm.adoc:13]: Could you use `dependency:org.apache.logging.log4j:log4j-api[version="2.13.3"]` for these snippets to support Maven too

File: elasticsearch/src/main/resources/META-INF/native-image/io.micronaut.elasticsearch/elasticsearch/reflect-config.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
+[
+  {
+    "name": "org.apache.commons.logging.LogFactory"
+  },
```
Comment (graemerocher) [inline elasticsearch/src/main/resources/META-INF/native-image/io.micronaut.elasticsearch/elasticsearch/reflect-config.json:4]: Maybe this is fine for now but I think long term we are going to need a `micronaut-logging` project with `micronaut-log4j` and `micronaut-log4j2` projects that contain this config and can be shared.

File: src/main/docs/guide/graalvm.adoc (hunk 1 above)
Comment (ilopmar) [reply src/main/docs/guide/graalvm.adoc]: done

File: elasticsearch/src/main/resources/META-INF/native-image/io.micronaut.elasticsearch/elasticsearch/reflect-config.json (hunk 2 above)
Comment (ilopmar) [reply elasticsearch/src/main/resources/META-INF/native-image/io.micronaut.elasticsearch/elasticsearch/reflect-config.json:4]: Agree

**Pull request merged**